pwiz_tools/Skyline/FileUI/PeptideSearch/BuildPeptideSearchLibraryControl.cs

```diff
@@ -280,12 +305,32 @@ public bool BuildOrUsePeptideSearchLibrary(CancelEventArgs e)
             {
                 return AddExistingLibrary(e);
             }
-            else
+            else if (PerformDDASearch)
+            {
+                return PresetNecessarySettingsWithoutLibrary(e);
+            } else 
             {
                 return BuildPeptideSearchLibrary(e);
             }
         }
 
+        private bool PresetNecessarySettingsWithoutLibrary(CancelEventArgs cancelEventArgs)
+        {
+            //ImportPeptideSearch.DocLib = new EmptyLibraryForDDASearch();
+            foreach (string rawFile in ImportPeptideSearch.SearchFilenames)
+            {
+                if (!Path.GetExtension(rawFile).ToUpper().Equals(".MGF"))
+                    ImportPeptideSearch.SpectrumSourceFiles.Add(rawFile, new ImportPeptideSearch.FoundResultsFilePossibilities(rawFile){ExactMatch = rawFile});

```

Comment:
Am I understanding correctly that with the Perform DDA Search option, FoundResultsFiles indicates the search engine source files (mzML, RAW, MGF, etc)?

Reply:
It could mean that, but I think it is trying for the files which Skyline will extract chromatograms from, which usually are not MGF files and may not even be mzXML files, if the searched mzXML files had only MS/MS spectra, as was once the case in the Aebersold lab.

Reply:
yes, the search engine searches all files in the SearchFileNames list but already uses these files to indicate Skyline where to extract the chromatograms from. After the search is finished, the SpectrumSourceFiles are replaced by the correct search result and the original source file. We might have to extend this if clause by other extensions, though... I don't know which file formats are fine to extract chromatograms... shall I add the mzXML here?

Reply:
I don't see any way to do this reliably with only file extension. Sure, MGF will never have MS1s, but most other open formats might or might not. I don't see any way to tell whether those contain the MS1s other than opening them to check. Simpler would be to use the searched files for XIC extraction only if a vendor raw file with the same basename can't be found nearby (not sure how this would work for WIFF though). Then if the searched file ends up being used, if it happens to only contain MS2s, a suitable error message can be given.